crates/nu-command/src/filters/rotate/counter_clockwise.rs

```diff
@@ -0,0 +1,213 @@
+                } => {
+                    old_column_names = cols;
+                    for v in vals {
+                        new_values.push(v)
+                    }
+                }
+                Value::List { vals, span: _ } => {
+                    not_a_record = true;
+                    for v in vals {
+                        new_values.push(v);
+                    }
+                }
+                Value::String { val, span } => {
+                    old_column_names = vec!["".to_string()];
+                    not_a_record = true;
+                    new_values.push(Value::String { val, span })
+                }
+                x => {
+                    old_column_names = vec!["".to_string()];
+                    not_a_record = true;
+                    new_values.push(x)
+                }
+            }
+        }
+    } else {
+        return Err(ShellError::UnsupportedInput(
+            "Rotate command requires a Nu value as input".to_string(),
+            call.head,
+        ));
+    }
+
+    let total_columns = &old_column_names.len();
+
+    // holder for the new column names, particularly if none are provided by the user we create names as Column0, Column1, etc.
+    let mut new_column_names = {
+        let mut res = vec![];
+        for idx in 0..(total_rows + total_columns) - 1 {
+            res.push(format!("Column{}", idx));
+        }
+        res.to_vec()
+    };
+
+    // we got new names for columns from the input, so we need to swap those we already made
+    if !col_given_names.is_empty() {
+        for (idx, val) in col_given_names.into_iter().enumerate() {
+            if idx > new_column_names.len() - 1 {
+                break;
+            }
+            new_column_names[idx] = val;
+        }
+    }
+
+    if not_a_record {
+        return Ok(Value::List {
+            vals: vec![Value::Record {
+                cols: new_column_names,
+                vals: new_values,
+                span: call.head,
+            }],
+            span: call.head,
+        }
+        .into_pipeline_data());
+    }
+
+    // holder for the new records
+    let mut final_values = vec![];
+
+    // the number of initial columns will be our number of rows, so we iterate through that to get the new number of rows that we need to make
+    // as we're rotating counter clockwise, we're iterating from right to left
+    for (idx, val) in old_column_names.iter().enumerate().rev() {
+        let mut res = vec![Value::String {
+            val: val.to_string(),
+            span: call.head,
+        }];
+
+        let new_vals = {
+            // move through the array every 2 elements, starting from our old column's index
+            // so if initial data was like this [[a b]; [1 2] [3 4]] - we basically iterate on this [1 2 3 4] array, so we pick 2, then 4, and then when idx decreases (notice the .rev()), we pick 1 and 3
+            for i in (idx..new_values.len()).step_by(2) {
```

Comment:
should this be based on the size of the table? If you had a 3x3 table, or say a 2x3 table, will this also work?

Reply:
good catch, let me see.

Reply:
ok, the step has to be `new_values.len() / total_rows`. I fixed it and pushing it now.